syntax = "proto3";

service ChassisService {
    rpc Drive(stream ChassisData) returns (stream ChassisFeedback);
}

message ChassisData {
    double velocity = 1;
    double rotation = 2;
}

message ChassisFeedback {
    int32 leftPower = 1;
    int32 rightPower = 2;
    int32 errorCode = 5;
    string errorDescription = 6;
}